src/jesse_validator_draft6.erl

```diff
@@ -978,18 +978,23 @@ check_format(Value, _Format = <<"uri">>, State) when is_binary(Value) ->
   %% not yet supported
   State;
 check_format(Value, <<"uri-reference">>, State) when is_binary(Value) ->
-  State;
-  %%
-  %% The following code works for OTP 21+ but fails for OTP 19-20.
-  %%
-  %% case uri_string:parse(Value) of
-  %%   {error, _ErrorType, _Term} ->
-  %%     handle_data_invalid(?wrong_format, Value, State);
-  %%   _ -> State
-  %% end;
+  uri_reference(Value, State);
 check_format(_Value, _Format, State) ->
   State.
 
+-ifdef(OTP_RELEASE).
+uri_reference(Value, State) when is_binary(Value) ->
+  case uri_string:parse(Value) of
+    {error, _ErrorType, _Term} ->
+      handle_data_invalid(?wrong_format, Value, State);
+    _ -> State
+  end.
+-else.
+uri_reference(_Value, State) ->
+  State.
```

Comment:
and for pre-OTP-21 you can try to use `http_uri` module https://erldocs.com/20.0/inets/http_uri.html?i=0&search=http#undefined. But I think it's not that important.